### ISSUE TITLE
Occasionally responds to random messages even with though participationChance is set to 0.0

### ISSUE THREAD

**Issue by SystemPunch** [clux/cleverbot-irc#11]
I had the bot in a channel last night with participationChance set to 0.0 and noticed that it would still occasionally respond to random messages, much to the annoyance of the person it was responding to

Not sure of a way to reproduce, other than setting to 0.0 and letting it loose in a channel for a couple of hours

I haven't tried it with just 0, but maybe I should just set it to a negative number since trials.bernoulli() just does a comparison between Math.random() and participationChance

**Comment by clux** [comment]
That really should not happen. `Math.random()` returns in the range `[0, 1)` and it checks against `< 0` in your case.

Did a sanity now with `for (var i =0; i<1000000000; i++) { if (trials.bernoulli(0)) console.log('failed' + i); }`

The only thing I can thing about is that you accidentally said a sentence that started with the name of the bot without meaning to talk to it. That's the only way [irc-stream](https://github.com/clux/irc-stream/blob/8caa9c1002a019ff08ddd3a3063c7d5b3de8586e/irc-stream.js#L47-L52) to let it through.

If it's not that, and you have a line that fails, I would like to see it.

**Comment by SystemPunch** [comment]
Yeah, I was also really confused when I saw sentences pop up in the console that didn't start with the bot's name (that was also the first thing I thought of) and when I do the same sanity check it always goes as expected

For now, I'm happy to just forget about it. Might tweak irc-stream to log the bernoulli trial results so that if it happens again I can have some evidence

**Comment by clux** [comment]
It is possible that it's just a network like issue as well. The callback from `clvr` can happen much later than you may think, and I have actually seen consistent several minute++ long response times on `wolfram` at the time when a hypervisor was having issues. May lead you to misidentify what clvr's message is in response to.

At any rate, if you do find anything with tweaks/monitoring, keep me posted. Should iron it out if it's a real issue.
